sale_commission_oca/models/sale_order.py

```diff
@@ -54,17 +55,20 @@ class SaleOrderLine(models.Model):
 
     @api.depends("order_id.partner_id")
     def _compute_agent_ids(self):
-        self.agent_ids = False  # for resetting previous agents
         for record in self:
+            agent_ids = False  # for resetting previous agents
```

Comment:
Name the variables correctly:

```suggestion
            agent_vals = False
```

Anyway, why this change? It was correct and performant previously. There's no problem in assigning several times the field inside the method.

Reply:
Hi @pedrobaeza,
While strictly functional, it triggers a double assignment for matching records.
So, It dirties the cache twice.
It can potentially triggers downstream @api.depends computations twice (once for the reset, once for the value).
I refactored it to ensure a single write per record to optimize the ORM cache usage.

Reply:
Not really, as the computation is deferred until needed, or when saving, but not on intermediate steps.

Reply:
All right then, I'll leave it as it was.